main.go

```diff
@@ -52,80 +52,123 @@ var (
 	reportFile       = "report.txt"
 	dbNameOverride   string
 	instanceOverride string
+	projectOverride  string
 	filePrefix       = ""
 	verbose          bool
 )
 
+func init() {
+	flag.StringVar(&dbNameOverride, "dbname", "", "dbname: name to use for Spanner db")
+	flag.StringVar(&instanceOverride, "instance", "", "instance: Spanner instance to use")
+	flag.StringVar(&projectOverride, "project", "", "project: Google Cloud project to use")
+	flag.StringVar(&filePrefix, "prefix", "", "prefix: file prefix for generated files")
+	flag.BoolVar(&verbose, "v", false, "verbose: print additional output")
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+	flag.PrintDefaults()
+	fmt.Fprintf(os.Stderr, `Note: input is always read from stdin.
+Sample usage:
+  pg_dump mydb | %s
+  %s < my_pg_dump_file
+`, os.Args[0], os.Args[0])
+}
+
 func main() {
 	flag.Usage = usage
 	flag.Parse()
 	internal.VerboseInit(verbose)
-	now := time.Now()
 	lf, err := setupLogFile()
 	if err != nil {
 		fmt.Printf("\nCan't set up log file: %v\n", err)
 		panic(fmt.Errorf("can't set up log file"))
 	}
 	defer close(lf)
-	project, err := getProject()
+
+	err = process(projectOverride, instanceOverride, dbNameOverride, nil, filePrefix)
 	if err != nil {
-		fmt.Printf("\nCan't get project: %v\n", err)
```

Comment:
I agree that doing the printf and then the panic looks kind of redundant, but it helps generate a readable error message.

Currently, this prints something like:

    Generating schema: 100%
    Wrote schema to file 'pg_dump_2020-02-19_817a-c14b.schema.txt'.
    Creating new database pg_dump_2020-02-19_817a-c14b in instance e2e-spanner-testing with default permissions ... 
    Can't create database: can't build CreateDatabaseRequest: rpc error: code = Unauthenticated desc = transport: oauth2: cannot fetch token: 400 Bad Request
    Response: {
      "error": "invalid_grant",
      "error_description": "reauth related error (invalid_rapt)",
      "error_subtype": "invalid_rapt"
    }.
    
    Possible cause: credentials are mis-configured. Do you need to run
    
      gcloud auth application-default login
    
    or configure environment variable GOOGLE_APPLICATION_CREDENTIALS.
    See https://cloud.google.com/docs/authentication/getting-started.
    
    panic: can't create database

    goroutine 1 [running]:
    main.main()
    	/usr/local/google/home/nch/cloud-spanner/postgres-to-spanner/harbourbridge/main.go:102 +0xd73
    exit status 2

But switching to just the panic produces:
    
    Generating schema: 100%
    Wrote schema to file 'pg_dump_2020-02-19_6248-fecd.schema.txt'.
    Creating new database pg_dump_2020-02-19_6248-fecd in instance e2e-spanner-testing with default permissions ... panic: can't create database: can't build CreateDatabaseRequest: rpc error: code = Unauthenticated desc = transport: oauth2: cannot fetch token: 400 Bad Request
    Response: {
      "error": "invalid_grant",
      "error_description": "reauth related error (invalid_rapt)",
      "error_subtype": "invalid_rapt"
    }.
    
    Possible cause: credentials are mis-configured. Do you need to run
    
      gcloud auth application-default login
    
    or configure environment variable GOOGLE_APPLICATION_CREDENTIALS.
    See https://cloud.google.com/docs/authentication/getting-started.
    
    
    goroutine 1 [running]:
    main.main()
    	/usr/local/google/home/nch/cloud-spanner/postgres-to-spanner/harbourbridge/main.go:103 +0xd26
    exit status 2

which is harder to read and figure out what went wrong.

What we should really do is use log.Fatal, but because of the noise of the client library (it prints out spammy logs), we probably don't want to switch to that right now.